src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure.Core/Configurations/GCPerfSimFunctional.Configuration.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GC.Infrastructure.Core.Configurations
+{
+    public sealed class GCPerfSimFunctionalConfiguration
+    {
+        public string output_path { get; set; }
+        public string gcperfsim_path { get; set; }
+        public Dictionary<string, CoreRunInfo> coreruns { get; set; }
+        public Dictionary<string, string>? environment_variables { get; set; }
+        public string trace_configuration_type { get; set; } = "gc";
+    }
+
+    public static class GCPerfSimFunctionalConfigurationParser
+    {
+        public static GCPerfSimFunctionalConfiguration Parse(string path, bool isIncompleteConfiguration = false)
+        {
+            string serializedConfiguration = File.ReadAllText(path);
+            GCPerfSimFunctionalConfiguration configuration = Common.Deserializer.Deserialize<GCPerfSimFunctionalConfiguration>(serializedConfiguration);
+
+            // Preconditions.
+            if (configuration.coreruns == null)
+            {
+                throw new ArgumentException($"{nameof(GCPerfSimFunctionalConfigurationParser)}: Provide a set of coreruns use for the analysis.");
+            }
+
+            if (string.IsNullOrEmpty(configuration.output_path))
+            {
+                throw new ArgumentException($"{nameof(GCPerfSimFunctionalConfigurationParser)}: Provide an output path.");
+            }
+
+            // The rest of the items aren't filled for the incomplete configuration that's programmatically filled by the infrastructure.
+            if (isIncompleteConfiguration)
```

Comment:
This isn't needed - get rid of the isIncompleteConfiguration functionality as it's not needed.